trying to put the verses

Notes
======
trying to put the verses

Environment
======
Environment Key | Value
:----: | :----:
version | 9.0
build | 138
UDID | 1d8223c4912d8843
Android Release | 4.4.4
Android SDK | 19
Brand | samsung
Device | gtel3g
Stack trace
======
```java
java.lang.StringIndexOutOfBoundsException
	at android.text.SpannableStringBuilder.<init>(SpannableStringBuilder.java:54)
	at android.text.SpannableStringBuilder.subSequence(SpannableStringBuilder.java:905)
	at com.door43.translationstudio.newui.translate.ReviewModeAdapter$23$1.onDrag(ReviewModeAdapter.java:1420)
	at android.view.View.dispatchDragEvent(View.java:17624)
	at android.view.ViewGroup.notifyChildOfDrag(ViewGroup.java:1333)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1175)
	at android.view.ViewGroup.notifyChildOfDrag(ViewGroup.java:1333)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1175)
	at android.view.ViewGroup.notifyChildOfDrag(ViewGroup.java:1333)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1175)
	at android.view.ViewGroup.notifyChildOfDrag(ViewGroup.java:1333)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1175)
	at android.view.ViewGroup.notifyChildOfDrag(ViewGroup.java:1333)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1175)
	at android.view.ViewGroup.notifyChildOfDrag(ViewGroup.java:1333)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1175)
	at android.view.ViewGroup.notifyChildOfDrag(ViewGroup.java:1333)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1175)
	at android.view.ViewGroup.notifyChildOfDrag(ViewGroup.java:1333)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1175)
	at android.view.ViewGroup.notifyChildOfDrag(ViewGroup.java:1333)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1175)
	at android.view.ViewGroup.notifyChildOfDrag(ViewGroup.java:1333)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1175)
	at android.view.ViewGroup.notifyChildOfDrag(ViewGroup.java:1333)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1175)
	at android.view.ViewGroup.notifyChildOfDrag(ViewGroup.java:1333)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1175)
	at android.view.ViewGroup.notifyChildOfDrag(ViewGroup.java:1333)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1175)
	at android.view.ViewGroup.notifyChildOfDrag(ViewGroup.java:1333)
	at android.view.ViewGroup.dispatchDragEvent(ViewGroup.java:1175)
	at android.view.ViewRootImpl.handleDragEvent(ViewRootImpl.java:5244)
	at android.view.ViewRootImpl.access$900(ViewRootImpl.java:115)
	at android.view.ViewRootImpl$ViewRootHandler.handleMessage(ViewRootImpl.java:3456)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:136)
	at android.app.ActivityThread.main(ActivityThread.java:5590)
	at java.lang.reflect.Method.invokeNative(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:515)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1268)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1084)
	at dalvik.system.NativeStart.main(Native Method)

```
Log history
======
```java
7/28/16 12:07 PM  E/com.door43.translationstudio.tasks.CheckForLatestReleaseTask: Failed to check for the latest release
javax.net.ssl.SSLHandshakeException: javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0x6e83d008: Failure in SSL library, usually a protocol error
error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure (external/openssl/ssl/s23_clnt.c:744 0x68684ec8:0x00000000)
	at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:449)
	at com.android.okhttp.Connection.upgradeToTls(Connection.java:146)
	at com.android.okhttp.Connection.connect(Connection.java:107)
	at com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:294)
	at com.android.okhttp.internal.http.HttpEngine.sendSocketRequest(HttpEngine.java:255)
	at com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:206)
	at com.android.okhttp.internal.http.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:345)
	at com.android.okhttp.internal.http.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:296)
	at com.android.okhttp.internal.http.HttpURLConnectionImpl.getInputStream(HttpURLConnectionImpl.java:179)
	at com.android.okhttp.internal.http.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:246)
	at com.door43.tools.reporting.Github.getRequest(Github.java:52)
	at com.door43.tools.reporting.Github.getLatestRelease(Github.java:38)
	at com.door43.translationstudio.tasks.CheckForLatestReleaseTask.start(CheckForLatestReleaseTask.java:34)
	at com.door43.util.tasks.ManagedTask.run(ManagedTask.java:77)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:841)
Caused by: javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0x6e83d008: Failure in SSL library, usually a protocol error
error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure (external/openssl/ssl/s23_clnt.c:744 0x68684ec8:0x00000000)
	at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
	at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:406)
	... 16 more

7/28/16 11:27 AM  W/com.door43.translationstudio.core.TargetTranslation: Missing manifest file in target translation ajg_jhn_text_reg
7/28/16 11:27 AM  W/com.door43.translationstudio.core.TargetTranslation: Missing manifest file in target translation ajg_jhn_text_reg
7/28/16 9:11 AM  W/com.door43.translationstudio.core.TargetTranslation: Missing manifest file in target translation ajg_mat_text_reg
7/28/16 9:11 AM  W/com.door43.translationstudio.core.TargetTranslation: Missing manifest file in target translation ajg_mat_text_reg
     

```

--- a/app/src/main/java/com/door43/translationstudio/newui/translate/ReviewModeAdapter.java
+++ b/app/src/main/java/com/door43/translationstudio/newui/translate/ReviewModeAdapter.java
@@ -1493,9 +1493,11 @@ public class ReviewModeAdapter extends ViewModeAdapter<ReviewModeAdapter.ViewHol
                             if (event.getAction() == DragEvent.ACTION_DRAG_STARTED) {
                                 // delete old span
                                 int[] spanRange = (int[])event.getLocalState();
-                                CharSequence in = editText.getText();
-                                CharSequence out = TextUtils.concat(in.subSequence(0, spanRange[0]), in.subSequence(spanRange[1], in.length()));
-                                editText.setText(out);
+                                if((spanRange != null) && (spanRange.length >= 2) ) {
+                                    CharSequence in = editText.getText();
+                                    CharSequence out = TextUtils.concat(in.subSequence(0, spanRange[0]), in.subSequence(spanRange[1], in.length()));
+                                    editText.setText(out);
+                                }
                             } else if(event.getAction() == DragEvent.ACTION_DROP) {
                                 int offset = editText.getOffsetForPosition(event.getX(), event.getY());
                                 CharSequence text = editText.getText();

--- a/app/src/main/java/com/door43/translationstudio/newui/translate/ReviewModeAdapter.java
+++ b/app/src/main/java/com/door43/translationstudio/newui/translate/ReviewModeAdapter.java
@@ -1495,8 +1495,10 @@ public class ReviewModeAdapter extends ViewModeAdapter<ReviewModeAdapter.ViewHol
                                 int[] spanRange = (int[])event.getLocalState();
                                 if((spanRange != null) && (spanRange.length >= 2) ) {
                                     CharSequence in = editText.getText();
-                                    CharSequence out = TextUtils.concat(in.subSequence(0, spanRange[0]), in.subSequence(spanRange[1], in.length()));
-                                    editText.setText(out);
+                                    if( (spanRange[0] < in.length()) && spanRange[1] < in.length()) {
+                                        CharSequence out = TextUtils.concat(in.subSequence(0, spanRange[0]), in.subSequence(spanRange[1], in.length()));
+                                        editText.setText(out);
+                                    }
                                 }
                             } else if(event.getAction() == DragEvent.ACTION_DROP) {
                                 int offset = editText.getOffsetForPosition(event.getX(), event.getY());

